.github/workflows/publish-artifacts.yml

```diff
@@ -0,0 +1,173 @@
+name: Publish artifacts to ORAS
+
+on:
+  push:
+    branches:
+    - main
+
+env:
+  RUST_TOOLCHAIN: 1.80.0
+
+jobs:
+  publish-aa:
+    permissions:
+      contents: read
+      packages: write
+      id-token: write
+    strategy:
+      matrix:
+        tee:
+        - none
+        - amd
+        - az-cvm-vtpm
+        - tdx
+        - se
+        - cca
+        include:
+        - tee: none
+          libc: musl
+          arch: x86_64
+        - tee: amd
+          libc: musl
+          arch: x86_64
+        - tee: az-cvm-vtpm
+          libc: gnu
+          arch: x86_64
+        - tee: tdx
+          libc: gnu
+          arch: x86_64
+        - tee: se 
+          libc: gnu
+          arch: s390x
+        - tee: cca
+          libc: musl
+          arch: x86_64
+    runs-on: ${{ matrix.arch == 's390x' && 's390x' || 'ubuntu-22.04' }}
+    env:
+      TEE_PLATFORM: ${{ matrix.tee }}
+      LIBC: ${{ matrix.libc }}
+      REGISTRY: ghcr.io
+      IMAGE_NAME: ${{ github.repository }}
+    steps:
+    - name: Take a pre-action for self-hosted runner
+      if: matrix.arch == 's390x'
+      run: |
+        if [ -f "${HOME}/script/pre_action.sh" ]; then
+          "${HOME}/script/pre_action.sh" cc-guest-components
+        fi
+
+    - name: Log in to the Container registry
+      uses: docker/login-action@65b78e6e13532edd9afa3aa52ac7964289d1a9c1
+      with:
+        registry: ${{ env.REGISTRY }}
+        username: ${{ github.actor }}
+        password: ${{ secrets.GITHUB_TOKEN }}
+
+    - uses: oras-project/setup-oras@v1
+      with:
+        version: 1.0.0
```

Comment:
I think we need to use oras 1.2.0+ to get multi-arch support for the pull at least, I'm not sure about the push, but it might be worth updating it to align with kata and CAA's version anyway?